staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -507,13 +507,59 @@ func TestBackoffOnTooManyRequests(t *testing.T) {
 	}
 
 	stopCh := make(chan struct{})
-	r.ListAndWatch(stopCh)
+	if err := r.ListAndWatch(stopCh); err != nil {
+		t.Fatal(err)
+	}
 	close(stopCh)
 	if bm.calls != 2 {
 		t.Errorf("unexpected watch backoff calls: %d", bm.calls)
 	}
 }
 
+func TestNoRelistOnTooManyRequests(t *testing.T) {
+	err := apierrors.NewTooManyRequests("too many requests", 1)
+	clock := &clock.RealClock{}
+	bm := &fakeBackoff{clock: clock}
+	listCalls, watchCalls := 0, 0
+
+	lw := &testLW{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			listCalls++
+			return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "1"}}, nil
```

Comment:
I always thought that list requests would also be retried on a 429.
But I couldn't find the retry logic in client-go that would handle [this](https://github.com/kubernetes/kubernetes/blob/ad3d138cda76fc0267da5131fa3ff7906e2ddf76/staging/src/k8s.io/client-go/rest/request.go#L1138).

Reply:
It is retried - it's handled here:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/with_retry.go#L179-L192

Reply:
Ok, thanks.